cura/CuraPackageManager.py

```diff
@@ -0,0 +1,357 @@
+            self._to_remove_package_set = set(management_dict["to_remove"])
+            self._to_install_package_dict = management_dict["to_install"]
+
+            Logger.log("i", "Package management file %s is loaded", self._package_management_file_path)
+
+    def _saveManagementData(self) -> None:
+        with open(self._package_management_file_path, "w", encoding = "utf-8") as f:
+            data_dict = {"installed": self._installed_package_dict,
+                         "to_remove": list(self._to_remove_package_set),
+                         "to_install": self._to_install_package_dict}
+            data_dict["to_remove"] = list(data_dict["to_remove"])
+            json.dump(data_dict, f)
+            Logger.log("i", "Package management file %s is saved", self._package_management_file_path)
+
+    # (for initialize) Removes all packages that have been scheduled to be removed.
+    def _removeAllScheduledPackages(self) -> None:
+        print("Will purge", self._to_remove_package_set)
+        for package_id in self._to_remove_package_set:
+            self._purgePackage(package_id)
+        self._to_remove_package_set.clear()
+        self._saveManagementData()
+
+    # (for initialize) Installs all packages that have been scheduled to be installed.
+    def _installAllScheduledPackages(self) -> None:
+        for package_id, installation_package_data in self._to_install_package_dict.items():
+            self._installPackage(installation_package_data)
+        self._to_install_package_dict.clear()
+        self._saveManagementData()
+
+    # Checks the given package is installed. If so, return a dictionary that contains the package's information.
+    def getInstalledPackageInfo(self, package_id: str) -> Optional[dict]:
+        if package_id in self._to_remove_package_set:
+            return None
+
+        if package_id in self._to_install_package_dict:
+            package_info = self._to_install_package_dict[package_id]["package_info"]
+            package_info["is_bundled"] = False
+            return package_info
+
+        if package_id in self._installed_package_dict:
+            package_info = self._installed_package_dict.get(package_id)
+            package_info["is_bundled"] = False
+            return package_info
+
+        for section, packages in self.getAllInstalledPackagesInfo().items():
+            for package in packages:
+                if package["package_id"] == package_id:
+                    package_info = package
+                    return package_info
+
+        return None
+
+    def getAllInstalledPackagesInfo(self) -> dict:
+        installed_package_id_set = set(self._installed_package_dict.keys()) | set(self._to_install_package_dict.keys())
+        installed_package_id_set = installed_package_id_set.difference(self._to_remove_package_set)
+
+        managed_package_id_set = set(installed_package_id_set) | self._to_remove_package_set
```

Comment:
Don't need to cast `installed_package_id_set` to a set again. It's already a set.